repository-data/application/src/main/resources/hcm-config/configuration/update/DW-366-GdprUpdateWhoCanAccessField.groovy

```diff
@@ -0,0 +1,74 @@
+package org.hippoecm.frontend.plugins.cms.admin.updater
+
+import org.onehippo.repository.update.BaseNodeUpdateVisitor
+
+import javax.jcr.Node
+import javax.jcr.NodeIterator
+
+/**
+ * Document type updated whocanaccess field from String to Rich Text.
+ * This script copies data from old field to new, and removes old field
+ */
+class GdprUpdateWhoCanAccessField extends BaseNodeUpdateVisitor {
+
+    private static final String PROPERTY_GDPR_WHOCANACCESS_OLD = "website:whocanaccess"
+    private static final String PROPERTY_GDPR_WHOCANACCESS_NEW = "website:whocanaccessinfo"
+
+    boolean doUpdate(Node node) {
+
+        // Query returns the hippo:handle node for the document
+        // (which has the 3 variants)
+        try {
+            if (node.hasNodes()) {
+                return updateFieldAndSaveDocument(node)
+            }
+        } catch (e) {
+            log.error("Failed to process record.", e)
+        }
+
+        return false
+    }
+
+    boolean updateFieldAndSaveDocument(Node parentHippoHandleNode) {
+
+        NodeIterator nodeIterator = parentHippoHandleNode.getNodes()
+        int variantsUpdated = 0
+
+        // Iterate through each variant
+        while(nodeIterator.hasNext()) {
+            Node n = nodeIterator.nextNode()
+            String path = n.getPath()
+
+            if (!n.hasProperty(PROPERTY_GDPR_WHOCANACCESS_OLD)) {
+                continue
+            }
+
+            String currentValue
+            currentValue = n.getProperty(PROPERTY_GDPR_WHOCANACCESS_OLD).getString()
+
+            log.info("attempting to update node: " + path)
+
+            // Does node already have this new property?  If so skip.
+            if (n.hasNode(PROPERTY_GDPR_WHOCANACCESS_NEW)) {
+                log.info("New property already exists. Skipping : " + path)
+                continue
+            }
+
+            // create new html whocanaccess node property and assign current value
+            def newNode = n.addNode(PROPERTY_GDPR_WHOCANACCESS_NEW, "hippostd:html")
+            newNode.setProperty("hippostd:content", currentValue)
```

Comment:
Richtext fields by default write wrap their content with the `<p>` tag. If you want you can have a look if it's true in this case. In the worst case authors, while re-opeining a document modified by this script, will find this small change in the diff. Not something really impacting the PR, but if wrapping the currentValue with "`<p>`..`</p>`" can avoid it, maybe it could be a nice addition